playbooks/roles/nomad/tasks/csi_volumes.yml

```diff
@@ -0,0 +1,81 @@
+---
+
+- name: Check if volume has already been created and registered
+  ansible.builtin.uri:
+    url: "http://nomad.service.consul:4646/v1/volume/csi/{{ item.id }}"
+    method: GET
+    # headers:
+    #   X-Nomad-Token: "{{ nomad_token }}"
+    remote_src: yes
+  register: volume_exists
+  ignore_errors: true
+
+- name: Check variables
+  ansible.builtin.debug:
+    msg: |
+      Name: {{ item.name }}
+      Id: {{ item.id }}
+      PluginID: {{ nomad_csi_plugin_id }}
+
+      
+
+- name: Check for state of CSI Plugins to prevent race
+  block:
```

Comment:
Why is this a block? there is no condition on it?

Reply:
This was intended to use the `when: volume_exists.status == 404` to avoid running the tasks within the block if the volume already exists